### PR TITLE
[bundle-size] Create new API endpoint to receive bundleSizes as a JSON payload

### PR DESCRIPTION
**Pull request by danielrozenberg**
More changes included in this PR:
* Removes the text portion of the commit message now that the complete JSON is going to be included
* Reorganizes the tests into sub-sections using `describe`s. It might be easier to review these changes using a better diff program, such as Meld :) otherwise, trust that most of the test file changes is just shuffles (modulo required changes to the tests to cater to the aforementioned text removal), except for the addition of the `/commit/:headSha/store.json` test sub-section

### CODE REVIEW

File: bundle-size/app.js [hunk 1]
```diff
@@ -649,4 +649,61 @@ module.exports = app => {
 
     response.end();
   });
+
+  // TODO(danielrozenberg): replace the /store with this one, once the amphtml
+  // repo is in sync with this new change.
+  v0.post('/commit/:headSha/store.json', async (request, response) => {
```
Comment (rcebulko) [inline bundle-size/app.js]: I don't know if I've ever seen a POST URL with a file extension... :thinking:

File: bundle-size/app.js [hunk 2]
```diff
@@ -649,4 +649,61 @@ module.exports = app => {
...
+    }
+    if (
+      !bundleSizes ||
+      Object.values(bundleSizes).some(value => typeof value !== 'number')
+    ) {
+      return response
+        .status(400)
+        .end(
+          'POST request to /store must have a key/value object ' +
+            '(string->numeric) field "bundleSizes"'
```
Comment (rcebulko) [inline bundle-size/app.js]: Nit: `Map<string, number>` or similar JSDoc format might be clearer

File: bundle-size/app.js [hunk 3]
```diff
@@ -649,4 +649,61 @@ module.exports = app => {
...
+      app.log(
+        `Stored the new bundle size file bundle-size/${jsonBundleSizeFile} ` +
+          'the artifacts repository on GitHub'
+      );
+    } catch (error) {
+      const errorMessage =
+        `ERROR: Failed to create the bundle-size/${jsonBundleSizeFile} file ` +
+        'in the build artifacts repository on GitHub!\n' +
+        `Error message was: ${error}`;
+      app.log(errorMessage);
```
Comment (rcebulko) [inline bundle-size/app.js]: Nit: Use `app.log.error` or `app.log.warn` in this file where appropriate

File: bundle-size/app.js (hunk 1 above)
Comment (danielrozenberg) [reply bundle-size/app.js]: There's always a first time :P

File: bundle-size/app.js (hunk 3 above)
Comment (danielrozenberg) [reply bundle-size/app.js]: Neat

File: bundle-size/app.js (hunk 2 above)
Comment (danielrozenberg) [reply bundle-size/app.js]: Makes sense. Done.

**Pull request merged**